FiducialRegistrationWizard/Logic/vtkPointDistanceMatrix.cxx

```diff
@@ -0,0 +1,200 @@
+    return;
+  }
+
+  pointList1->DeepCopy( points );
+  UpdateNeededFlag = 1;
+}
+
+//------------------------------------------------------------------------------
+void vtkPointDistanceMatrix::SetPointList2( vtkPoints* points )
+{
+  if ( points == NULL )
+  {
+    vtkWarningMacro( "Input point list is null. Aborting set operation." );
+    return;
+  }
+
+  pointList2->DeepCopy( points );
+  UpdateNeededFlag = 1;
+}
+
+//------------------------------------------------------------------------------
+int vtkPointDistanceMatrix::GetPointList1Length()
+{
+  return pointList1->GetNumberOfPoints();
+}
+
+//------------------------------------------------------------------------------
+int vtkPointDistanceMatrix::GetPointList2Length()
+{
+  return pointList2->GetNumberOfPoints();
+}
+
+//------------------------------------------------------------------------------
+double vtkPointDistanceMatrix::GetElement( int pointList1Index, int pointList2Index )
+{
+  int pointList1Length = GetPointList1Length();
+  if ( pointList1Index >= pointList1Length )
+  {
+    vtkWarningMacro( "Point of list 1 is outside the range. Returning 0." )
+    return 0;
+  }
+
+  int pointList2Length = GetPointList2Length();
+  if ( pointList2Index >= pointList2Length )
+  {
+    vtkWarningMacro( "Point of list 2 is outside the range. Returning 0." )
+    return 0;
+  }
+
+  if ( UpdateNeededFlag  )
+  {
+    Update();
+  }
+
+  return distanceMatrix->GetComponent( pointList1Index, pointList2Index );
+}
+
+//------------------------------------------------------------------------------
+double vtkPointDistanceMatrix::GetMinimumElement()
+{
+  if ( distanceMatrix->GetNumberOfTuples() == 0 )
+  {
+    vtkGenericWarningMacro( "Matrix has no contents. Returning 0." )
+    return 0;
+  }
+
+  int pointList1Length = GetPointList1Length();
+  int pointList2Length = GetPointList2Length();
+  double minElement = GetElement( 0, 0 );
+  for ( int pointList1Index = 0; pointList1Index < pointList1Length; pointList1Index++ )
+  {
+    for ( int pointList2Index = 0; pointList2Index < pointList2Length; pointList2Index++ )
+    {
+      double currentElement = GetElement( pointList1Index, pointList2Index );
+      if ( currentElement < minElement )
+      {
+        minElement = currentElement;
+      }
+    }
+  }
+  return minElement;
+}
+
+//------------------------------------------------------------------------------
+void vtkPointDistanceMatrix::Update()
+{
+  int pointList2Length = GetPointList2Length();
+  distanceMatrix->SetNumberOfComponents( pointList2Length );
+  int pointList1Length = GetPointList1Length();
+  distanceMatrix->SetNumberOfTuples( pointList1Length );
+  for ( int pointList1Index = 0; pointList1Index < pointList1Length; pointList1Index++ )
+  {
+    for ( int pointList2Index = 0; pointList2Index < pointList2Length; pointList2Index++ )
+    {
+      double pointInList1[ 3 ] = { 0, 0, 0 }; // temporary values
```

Comment:
I wrote it out on multiple lines to make it more human-readable. But if the code works then I will reduce its length by using the this->PointList1->GetPoint(pointList1Index) and the Distance2BetweenPoints() functions